search-android/src/main/java/uk/os/elements/search/android/recentmanager/impl/provider/RecentsManagerImpl.java

```diff
@@ -0,0 +1,245 @@
+import rx.Subscriber;
+import rx.android.content.ContentObservable;
+import rx.functions.Func0;
+import rx.functions.Func1;
+import rx.schedulers.Schedulers;
+import rx.subjects.AsyncSubject;
+import uk.os.elements.search.SearchResult;
+import uk.os.elements.search.android.providers.bng.OsGridReference;
+import uk.os.elements.search.android.providers.latlon.LatLonResult;
+import uk.os.elements.search.android.providers.recents.RecentsManager;
+import uk.os.elements.search.android.recentmanager.impl.provider.content.searchresult.SearchResultColumns;
+import uk.os.elements.search.android.recentmanager.impl.provider.content.searchresult.SearchResultContentValues;
+import uk.os.elements.search.android.recentmanager.impl.provider.content.searchresult.SearchResultSelection;
+
+public class RecentsManagerImpl implements RecentsManager {
+
+    private final Context mContext;
+
+    public RecentsManagerImpl(Context context) {
+        mContext = context;
+    }
+
+    @Override
+    public Observable<List<SearchResult>> query(String query) {
+        final String term = sanitised(query);
+
+        Cursor cursor = mContext.getContentResolver().query(SearchResultColumns.CONTENT_URI, null,
+                null, null, SearchResultColumns.ACCESSED + " DESC");
+
+        if (cursor == null) {
+            List<SearchResult> empty = Collections.emptyList();
+            return Observable.just(empty);
+        }
+
+        return ContentObservable.fromCursor(cursor).map(new Func1<Cursor, SearchResult>() {
+            @Override
+            public SearchResult call(Cursor cursor) {
+                String featureId = cursor
+                        .getString(cursor.getColumnIndex(SearchResultColumns.FEATUREID));
+                String name = cursor.getString(cursor.getColumnIndex(SearchResultColumns.NAME));
+                String context = cursor
+                        .getString(cursor.getColumnIndex(SearchResultColumns.CONTEXT));
+                Double x = cursor.getDouble(cursor.getColumnIndex(SearchResultColumns.X));
+                Double y = cursor.getDouble(cursor.getColumnIndex(SearchResultColumns.Y));
+                int srid = cursor.getInt(cursor.getColumnIndex(SearchResultColumns.SRID));
+                Double minX = cursor.getDouble(cursor.getColumnIndex(SearchResultColumns.MINX));
+                Double minY = cursor.getDouble(cursor.getColumnIndex(SearchResultColumns.MINY));
+                Double maxX = cursor.getDouble(cursor.getColumnIndex(SearchResultColumns.MAXX));
+                Double maxY = cursor.getDouble(cursor.getColumnIndex(SearchResultColumns.MAXY));
+                return new SearchResult(featureId, name, context, new Point(x, y),
+                        new Envelope(minX, minY, maxX, maxY), SpatialReference.create(srid));
+            }
+        }).filter(new Func1<SearchResult, Boolean>() {
+            @Override
+            public Boolean call(SearchResult result) {
+                // TODO: improve
```

Comment:
Bit more detail re: improvement would be good here. Or maybe document elsewhere...